packages/gluestick/shared/lib/route-helper.js

```diff
@@ -5,6 +5,16 @@ import type { GetBeforeRoute } from '../types';
 
 const getBeforeRoute: GetBeforeRoute = (component = {}) => {
   const c: Object = component.WrappedComponent || component;
+
+  if (c.onEnter) {
+    return typeof window === 'undefined'
+      ? c.onEnter
+      : (...args) => {
+          c.onEnter(...args);
+          return Promise.resolve();
+        };
+  }
```

Comment:
I'm having trouble understanding this code. Can you explain this? I'm not sure why we aren't returning `return c.onEnter(...args)` but that is also because I don't entirely know what is happening.

Reply:
We do not return `c.onEnter(...args)` in browser to make transition between pages async/non-blocking, but to block rendering on server. Once we upgrade to RR4 this part will be gone and moved into `componentDidMount`.